src/main/java/com/mraof/minestuck/block/redstone/LogicGateBlock.java

```diff
@@ -0,0 +1,120 @@
+package com.mraof.minestuck.block.redstone;
+
+import com.mraof.minestuck.block.MSProperties;
+import com.mraof.minestuck.util.ParticlesAroundSolidBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.RedstoneDiodeBlock;
+import net.minecraft.particles.RedstoneParticleData;
+import net.minecraft.state.EnumProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.shapes.ISelectionContext;
+import net.minecraft.util.math.shapes.VoxelShape;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorldReader;
+import net.minecraft.world.World;
+
+import java.util.Random;
+
+public class LogicGateBlock extends RedstoneDiodeBlock
+{
+	public static final EnumProperty<LogicGateBlock.State> STATE = MSProperties.LOGIC_STATE;
```

Comment:
I forget how this is achieved, do I need to establish a Builder for new block properties?

Reply:
No, you don't need to create a builder for this. What I'm saying is that since you have one block for each logic type, you should handle the logic type in the same way that you handle `TimedSolidSwitchBlock.tickRate`.